tests/shopify_python/test_google_styleguide.py

```diff
@@ -225,3 +225,51 @@ def test_use_cond_exprs(self):
         # root_not_msg should add no messages
         with self.assertNoMessages():
             self.walk(root_not_msg)
+
+    def test_unary_lambda_func(self):
+        unary_root = astroid.builder.parse("""
+        def unaryfnc():
+            unary_pass = lambda x: not (x+3)
+            unary_msg = lambda x: -x
+        """)
+
+        # Unary Op Test 1: This will not trigger a message
+        ulam_pass = unary_root.body[0].body[0].value
+        with self.assertNoMessages():
+            self.walk(ulam_pass)
+
+        # Unary Op Test 2: This will trigger a message
+        ulam_msg = unary_root.body[0].body[1].value
+        unary_message = pylint.testutils.Message('lambda-func', node=ulam_msg, args={
+            'op': 'operator.neg',
+            'lamfun': 'lambda x: - x'
+            })
+        with self.assertAddsMessages(unary_message):
+            self.walk(ulam_msg)
+
+    def test_binary_lambda_func(self):
+        binary_root = astroid.builder.parse("""
+        def binaryfnc():
+            binary_pass = lambda x, y, z: x * y + z
+            binary_pass2 = lambda x: x + 3
+            binary_msg = lambda x, y: x * y
```

Comment:
As a more realistic example, can you re-write these as `some_var = reduce(operator.mul, iterable)`